From 0f7a971a796e04e9ef86d923ae343904a28a556b Mon Sep 17 00:00:00 2001
From: Zhangbin Tong <zebulun.tong@rock-chips.com>
Date: Tue, 14 Aug 2018 18:21:55 +0800
Subject: [PATCH] ARM: rockchip_defconfig: Enable CONFIG_DEBUG_SET_MODULE_RONX

Android P CDD document chapter "9.7. Security Features"

Kernel integrity and self-protection features are integral to Android
security. Device implementations:
[C-0-8] MUST implement strict kernel memory protections where executable
code is readonly, read-only data is non-executable and non-writable, and
writable data is nonexecutable (e.g. CONFIG_DEBUG_RODATA or CONFIG_STRICT_KERNEL_RWX ).

Change-Id: Id4deb57242b55c8b5de2ad7de8a48f49ffa3c3e3
Signed-off-by: Zhangbin Tong <zebulun.tong@rock-chips.com>
---
 arch/arm/configs/rockchip_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/configs/rockchip_defconfig b/arch/arm/configs/rockchip_defconfig
index fd1ef2e1466b..c2e531300fc5 100644
--- a/arch/arm/configs/rockchip_defconfig
+++ b/arch/arm/configs/rockchip_defconfig
@@ -696,6 +696,7 @@ CONFIG_SCHEDSTATS=y
 CONFIG_TIMER_STATS=y
 # CONFIG_DEBUG_PREEMPT is not set
 CONFIG_ENABLE_DEFAULT_TRACERS=y
+CONFIG_DEBUG_SET_MODULE_RONX=y
 CONFIG_KEYS=y
 CONFIG_SECURITY_PERF_EVENTS_RESTRICT=y
 CONFIG_SECURITY=y
-- 
2.35.3

